package/MDAnalysis/coordinates/base.py

```diff
@@ -2089,6 +2107,9 @@ class ReaderBase(ProtoReader):
        Provides kwargs to be passed to :class:`Timestep`
     .. versionchanged:: 1.0
        Removed deprecated flags functionality, use convert_units kwarg instead
+    .. versionchanged:: 2.0.0
+       Now supports (un)pickle. Upon unpickling,
+       the current timestep is retained by reconstrunction.
```

Comment:
Should we just move the `__getstate__` func into `ProtoReader`. I don't see the reason not to.

Reply:
I'm not sure here, I always saw ProtoReader as the thing that you fallback to when the ReaderBase implements something that just won't work. So there might be an argument that the current approach offers a bit more modularity, but putting it in ProtoReader would just make things cleaner.

Thoughts @MDAnalysis/coredevs ?

Reply:
I think it's good in ProtoReader – if it works there then it will be as widely available as possible.

Hopefully we get more readers to be tested with the reader api tests.